fem/transfer.hpp

```diff
@@ -220,18 +283,34 @@ class L2ProjectionGridTransfer : public GridTransfer
       // arrays. The entries of the i'th high-order element are stored at the
       // index given by offsets[i].
       mutable Array<real_t> R, P;
-      Array<int> offsets;
+      mutable Array<real_t> R_ea, P_ea;
+
+      const bool use_device, verify_solution;
 
    public:
       L2ProjectionL2Space(const FiniteElementSpace& fes_ho_,
-                          const FiniteElementSpace& fes_lor_);
+                          const FiniteElementSpace& fes_lor_,
+                          const bool use_device_,
+                          const bool verify_solution_,
+                          Coefficient* coeff_,
+                          MemoryType d_mt_ = MemoryType::HOST);
+
+      /*Same as above but assembles and stores R_ea, P_ea */
+      void DeviceL2ProjectionL2Space(const FiniteElementSpace& fes_ho_,
+                                     const FiniteElementSpace& fes_lor_,
+                                     const Coefficient* coeff_);
+
       /// Maps <tt>x</tt>, primal field coefficients defined on a coarse mesh
       /// with a higher order L2 finite element space, to <tt>y</tt>, primal
       /// field coefficients defined on a refined mesh with a low order L2
       /// finite element space. Refined mesh should be a uniform refinement of
       /// the coarse mesh. Coefficients are computed through minimization of L2
       /// error between the fields.
       void Mult(const Vector& x, Vector& y) const override;
+
+      //Perform mult on the device (same as above)
+      void DeviceMult(const Vector& x, Vector& y) const;
```

Comment:
As much as possible, it would be nice to not have separate code paths for the "device" version and "non-device" version. This is not always feasible since the device code can sometimes have extra limitations, but something to not. (In MFEM we do not usually have separate `DeviceMult` member functions)